wikispeedruns/lobbys.py

```diff
@@ -453,4 +453,24 @@ def get_lobby_anon_users(lobby_id: int):
         cursor.execute(query, (lobby_id,))
         results = cursor.fetchall()
 
-        return results
\ No newline at end of file
+        return results
+    
+
+def delete_lobby(lobby_id) -> bool:
+    setCheckOff = """
+    SET FOREIGN_KEY_CHECKS=OFF
+    """
+    query = """
+    DELETE FROM lobbys 
+    WHERE lobby_id=%s
+    """
+    setCheckOn= """
+    SET FOREIGN_KEY_CHECKS=ON
```

Comment:
This is a great start!!!! This change is a little complicated because we need to decide how we want to handle deleted lobbies. Currently this change would leave a lot of dangling references to deleted `lobby_id`s in `lobby_prompts` and `lobby_runs`. Writing out questions for myself/WikiSpeedruns Team.

We should probably delete all corresponding entires in the `user_lobbys` table. After that: 

Some options I can think of - 
1) Add a new `isHidden` column to `lobbys` table and just use that, no fiddling with `lobby_id`s
2) Set `lobby_id` to NULL in `lobbys` and/or `lobby_runs` and `lobby_prompts`
3) Delete entries from `lobbys` and corresponding runs and prompts from `lobby_runs` and `lobby_prompts`
4) Current implementation of just deleting entries from `lobbys` (in addition to `user_lobbys` as stated above) and just be fine with / handle dangling references
5) Some other combination of the above three or other ideas


@dqian3 @mliu59 thoughts?

Reply:
I like option 1, it's the easiest and keeps things from getting messed up. Furthermore, we don't have to worry about consistency issues if we add features to lobbies.

I think there's also some SQL magic to make this work and delete references (DELETE CASCADE), which could be another option.

Reply:
I think setting a lobby to hidden would be a good idea It'll also allow us to keep metrics related to lobby creation/deletion (what types of lobbies get deleted, etc.)

Reply:
I'd suggest we can probably keep most of this delete code for now just submitting the lobbys_api.py changes but without the delete button. and then have a separate pr renaming the button to 'archive' and set the lobby to hidden there, this also sets users expectations clearer on what is happening.

Reply:
> fancy SQL magic

Yeah we can do ON DELETE CASCADE to delete rows in other tables that reference the lobby or ON DELETE SET NULL to just null out the column in other tables